packages/web/src/common/store/cache/sagas.ts

```diff
@@ -241,99 +286,125 @@ export function* add(kind, entries, replace, persist) {
-    yield put(cacheActions.subscribe(kind, entriesToSubscribe))
+    yield* put(cacheActions.subscribe(kind, entriesToSubscribe))
   }
 }
 
 // Adds entries but first checks if they are confirming.
 // If they are, don't add or else we could be in an inconsistent state.
 function* watchAdd() {
-  yield takeEvery(cacheActions.ADD, function* (action) {
-    const { kind, entries, replace, persist } = action
-    yield call(add, kind, entries, replace, persist)
-  })
+  yield* takeEvery(
+    cacheActions.ADD,
+    function* (action: ReturnType<typeof cacheActions.add>) {
+      const { kind, entries, replace, persist } = action
+      add(kind, entries, replace, persist)
+      yield* call(add, kind, entries, replace, persist)
+    }
+  )
 }
 
 // Prune cache entries if there are no more subscribers.
 function* watchUnsubscribe() {
-  yield takeEvery(cacheActions.UNSUBSCRIBE, function* (action) {
-    const { kind, unsubscribers } = action
-
-    const cache = yield select(getCache, { kind })
-
-    // Remove all transitive subscriptions.
-    const transitiveSubscriptions = {} // keyed by Kind
-    unsubscribers.forEach((s) => {
-      const { id = cache.uids[s.uid] } = s
-      if (
-        id in cache.subscriptions &&
-        cache.subscribers[id] &&
-        cache.subscribers[id].size <= 1
-      ) {
-        cache.subscriptions[id].forEach((subscription) => {
-          if (!transitiveSubscriptions[subscription.kind]) {
-            transitiveSubscriptions[subscription.kind] = [
-              { uid: subscription.uid }
-            ]
-          } else {
-            transitiveSubscriptions[subscription.kind].push({
-              uid: subscription.uid
-            })
-          }
-        })
-      }
-    })
-    yield all(
-      Object.keys(transitiveSubscriptions).map((subscriptionKind) =>
-        put(
-          cacheActions.unsubscribe(
-            subscriptionKind,
-            transitiveSubscriptions[subscriptionKind]
+  yield* takeEvery(
+    cacheActions.UNSUBSCRIBE,
+    function* (action: ReturnType<typeof cacheActions.unsubscribe>) {
+      const { kind, unsubscribers } = action
+
+      const cache = yield* select(getCache, { kind })
+
+      // Remove all transitive subscriptions.
+      const transitiveSubscriptions: {
+        [key in Kind]?: {
+          uid: string
+          id?: number | undefined
+        }[]
+      } = {} // keyed by Kind
+      unsubscribers.forEach((s) => {
+        const { id = cache.uids[s.uid] } = s
+        if (
+          id in cache.subscriptions &&
+          cache.subscribers[id] &&
+          cache.subscribers[id].size <= 1
+        ) {
+          cache.subscriptions[id].forEach((subscription) => {
+            if (!transitiveSubscriptions[subscription.kind]) {
+              transitiveSubscriptions[subscription.kind] = [
+                { uid: subscription.uid }
+              ]
+            } else {
+              transitiveSubscriptions[subscription.kind]!.push({
```

Comment:
functional change: `!`. Should be always non-null because of the check on line 330